File: ios-client/StripeApplication/StripeApplication/Components/BankAccountList.swift
import SwiftUI

struct BankAccountList: View {
    let externalAccounts: [ExternalAccount]
    let isLoading: Bool
    let onAddBankAccount: () -> Void
    let onSetDefault: (String) -> Void
    let onDelete: (String) -> Void

    var body: some View {
        VStack(alignment: .leading, spacing: 12) {
            HStack {
                Text("Bank Accounts")
                    .font(.headline)
                Spacer()
                Button {
                    onAddBankAccount()
                } label: {
                    Image(systemName: "plus")
                        .foregroundStyle(.blue)
                }
                .disabled(isLoading)
            }

            if externalAccounts.isEmpty {
                Text("No bank accounts linked")
                    .foregroundStyle(.secondary)
                    .font(.subheadline)
            } else {
                ForEach(externalAccounts) { account in
                    BankAccountItem(
                        account: account,
                        isLoading: isLoading,
                        onSetDefault: { onSetDefault(account.id) },
                        onDelete: { onDelete(account.id) }
                    )
                }
            }
        }
        .padding()
        .background(
            RoundedRectangle(cornerRadius: 12)
                .fill(Color(.systemBackground))
                .shadow(color: .black.opacity(0.1), radius: 4, y: 2)
        )
    }
}

struct BankAccountItem: View {
    let account: ExternalAccount
    let isLoading: Bool
    let onSetDefault: () -> Void
    let onDelete: () -> Void

    @State private var showDeleteDialog = false

    var body: some View {
        HStack {
            Text("\u{1F3E6}")
                .font(.title2)

            VStack(alignment: .leading) {
                HStack(spacing: 8) {
                    Text("\(account.bankName ?? "Bank") \u{2022}\u{2022}\u{2022}\u{2022} \(account.last4)")
                        .font(.subheadline)
                        .fontWeight(.medium)
                    if account.defaultForCurrency {
                        BadgeView(text: "Default", color: .green)
                    }
                }
                Text("\(account.currency.uppercased()) \u{2022} \(account.country)")
                    .font(.caption)
                    .foregroundStyle(.secondary)
            }

            Spacer()

            if !account.defaultForCurrency {
                Button {
                    onSetDefault()
                } label: {
                    Image(systemName: "star")
                        .foregroundStyle(.blue)
                }
                .disabled(isLoading)
            }

            Button(role: .destructive) {
                showDeleteDialog = true
            } label: {
                Image(systemName: "trash")
                    .foregroundStyle(.red)
            }
            .disabled(isLoading)
        }
        .padding(12)
        .background(
            RoundedRectangle(cornerRadius: 8)
                .fill(Color(.secondarySystemBackground))
        )
        .alert("Remove Bank Account", isPresented: $showDeleteDialog) {
            Button("Cancel", role: .cancel) {}
            Button("Remove", role: .destructive) {
                onDelete()
            }
        } message: {
            Text("Are you sure you want to remove this bank account?")
        }
    }
}

File: android-client/app/src/main/java/com/example/stripedemo/features/bankaccounts/components/BankAccountList.kt
package com.example.stripedemo.features.bankaccounts.components

import androidx.compose.foundation.layout.*
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Add
import androidx.compose.material.icons.filled.Delete
import androidx.compose.material.icons.filled.Star
import androidx.compose.material.icons.outlined.Star
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.example.stripedemo.data.models.ExternalAccount

@Composable
fun BankAccountList(
    externalAccounts: List<ExternalAccount>,
    isLoading: Boolean,
    onAddBankAccount: () -> Unit,
    onSetDefault: (String) -> Unit,
    onDelete: (String) -> Unit,
    modifier: Modifier = Modifier
) {
    Card(
        modifier = modifier.fillMaxWidth()
    ) {
        Column(
            modifier = Modifier.padding(16.dp),
            verticalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Text(
                    text = "Bank Accounts",
                    style = MaterialTheme.typography.titleMedium,
                    fontWeight = FontWeight.Bold
                )

                IconButton(
                    onClick = onAddBankAccount,
                    enabled = !isLoading
                ) {
                    Icon(
                        imageVector = Icons.Default.Add,
                        contentDescription = "Add Bank Account",
                        tint = MaterialTheme.colorScheme.primary
                    )
                }
            }

            if (externalAccounts.isEmpty()) {
                Text(
                    text = "No bank accounts linked",
                    style = MaterialTheme.typography.bodyMedium,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
            } else {
                externalAccounts.forEach { account ->
                    BankAccountItem(
                        account = account,
                        isLoading = isLoading,
                        onSetDefault = { onSetDefault(account.id) },
                        onDelete = { onDelete(account.id) }
                    )
                }
            }
        }
    }
}

@Composable
private fun BankAccountItem(
    account: ExternalAccount,
    isLoading: Boolean,
    onSetDefault: () -> Unit,
    onDelete: () -> Unit,
    modifier: Modifier = Modifier
) {
    var showDeleteDialog by remember { mutableStateOf(false) }

    Card(
        modifier = modifier.fillMaxWidth(),
        colors = CardDefaults.cardColors(
            containerColor = MaterialTheme.colorScheme.surfaceVariant
        )
    ) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(12.dp),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically
        ) {
            Row(
                horizontalArrangement = Arrangement.spacedBy(12.dp),
                verticalAlignment = Alignment.CenterVertically
            ) {
                Text(
                    text = "🏦",
                    style = MaterialTheme.typography.headlineSmall
                )

                Column {
                    Row(
                        horizontalArrangement = Arrangement.spacedBy(8.dp),
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Text(
                            text = "${account.bankName ?: "Bank"} •••• ${account.last4}",
                            style = MaterialTheme.typography.bodyLarge,
                            fontWeight = FontWeight.Medium
                        )
                        if (account.defaultForCurrency) {
                            SuggestionChip(
                                onClick = {},
                                label = { Text("Default") },
                                modifier = Modifier.height(24.dp)
                            )
                        }
                    }
                    Text(
                        text = "${account.currency.uppercase()} • ${account.country}",
                        style = MaterialTheme.typography.bodySmall,
                        color = MaterialTheme.colorScheme.onSurfaceVariant
                    )
                }
            }

            Row {
                if (!account.defaultForCurrency) {
                    IconButton(
                        onClick = onSetDefault,
                        enabled = !isLoading
                    ) {
                        Icon(
                            imageVector = Icons.Outlined.Star,
                            contentDescription = "Set as Default",
                            tint = MaterialTheme.colorScheme.primary
                        )
                    }
                }

                IconButton(
                    onClick = { showDeleteDialog = true },
                    enabled = !isLoading
                ) {
                    Icon(
                        imageVector = Icons.Default.Delete,
                        contentDescription = "Delete",
                        tint = MaterialTheme.colorScheme.error
                    )
                }
            }
        }
    }

    if (showDeleteDialog) {
        AlertDialog(
            onDismissRequest = { showDeleteDialog = false },
            title = { Text("Remove Bank Account") },
            text = { Text("Are you sure you want to remove this bank account?") },
            confirmButton = {
                TextButton(
                    onClick = {
                        showDeleteDialog = false
                        onDelete()
                    },
                    colors = ButtonDefaults.textButtonColors(
                        contentColor = MaterialTheme.colorScheme.error
                    )
                ) {
                    Text("Remove")
                }
            },
            dismissButton = {
                TextButton(onClick = { showDeleteDialog = false }) {
                    Text("Cancel")
                }
            }
        )
    }
}
